doc/sphinx/language/extensions/arguments-command.rst

```diff
@@ -0,0 +1,440 @@
+
+     `!`
+       the function will be unfolded only if all the arguments marked with `!`
+       evaulate to constructors.  See :ref:`Args_effect_on_unfolding`.
+
+     :n:`@name {? % @scope }`
+       a *formal parameter* of the function :n:`smart_qualid` (i.e.
+       the parameter name used in the function definition).  Unless `rename` is specified,
+       the list of :n:`@name`\s must be a prefix of the formal parameters, including all implicit
+       arguments.  `_` can be used to skip over a formal parameter.
+       :token:`scope` can be either a scope name or its delimiting key.  See :ref:`binding_to_scope`.
+
+     `clear implicits`
+       makes all implicit arguments into explicit arguments
+     `default implicits`
+       automatically determine the implicit arguments of the object.
+       See :ref:`auto_decl_implicit_args`.
+     `rename`
+       rename implicit arguments for the object.  See the example :ref:`here <renaming_implicit_arguments>`.
+     `assert`
+       assert that the object has the expected number of arguments with the
+       expected names.  See the example here: :ref:`renaming_implicit_arguments`.
+     `clear scopes`
+       clears argument scopes of :n:`@smart_qualid`
+     `extra scopes`
+       defines extra argument scopes, to be used in case of coercion to ``Funclass``
+       (see the :ref:`implicitcoercions` chapter) or with a computed type.
+     `simpl nomatch`
+       prevents performing a simplification step for :n:`@smart_qualid`
+       that would expose a match construct in the head position.  See :ref:`Args_effect_on_unfolding`.
+     `simpl never`
+       prevents performing a simplification step for :n:`@smart_qualid`.  See :ref:`Args_effect_on_unfolding`.
+
+     `clear bidirectionality hint`
+       removes the bidirectionality hint, the `&`
+
+     :n:`@implicits_alt`
+       use to specify alternative implicit argument declarations
+       for functions that can only be
+       applied to a fixed number of arguments (excluding, for instance,
+       functions whose type is polymorphic).
+       For parsing, the longest list of implicit arguments matching the function application
+       is used to select which implicit arguments are inserted.
+       For printing, the alternative with the most implicit arguments is used; the
+       implict arguments will be omitted if :flag:`Printing Implicit` is not set.
+       See the example :ref:`here<example_more_implicits>`.
+
+       .. todo the above feature seems a bit unnatural and doesn't play well with partial
+          application.  See https://github.com/coq/coq/pull/11718#discussion_r408841762
+
+   Use :cmd:`About` to view the current implicit arguments setting for a :token:`smart_qualid`.
+
+   Or use the :cmd:`Print Implicit` command to see the implicit arguments
+   of an object (see :ref:`displaying-implicit-args`).
+
+.. todo: indent the messages, example and note below?
```

Comment:
I'd rather suggest the following:
- move the error message about `/` in the description of this syntax above;
- same thing for `&`;
- move the warning about `assert` in the description of this clause above;
- create a new sub-section title before the first `.. example::` and call it "Manual declaration of implicit arguments" (by opposition to "Automatic declaration of implicit arguments").

Reply:
FWIW "in constrast to" would be better than "by opposition to".